src/widget.cpp

```diff
@@ -770,11 +770,10 @@ NWidgetBase::NWidgetBase(WidgetType tp) : ZeroedMemoryAllocator()
  * Mark the widget as 'dirty' (in need of repaint).
  * @param w Window owning the widget.
  */
-void NWidgetBase::SetDirty(const Window *w) const
+void NWidgetBase::SetDirty(Window *w)
 {
-	int abs_left = w->left + this->pos_x;
-	int abs_top = w->top + this->pos_y;
-	SetDirtyBlocks(abs_left, abs_top, abs_left + this->current_x, abs_top + this->current_y);
+	this->is_dirty = true;
+	w->flags |= WF_DIRTY;
```

Comment:
Do we need to be marking the entire window dirty here, or should it be `WF_WIDGETS_DIRTY`; or have I misunderstood the situation completely?